### ISSUE TITLE
Can db.find() return a value?

### ISSUE THREAD

**Issue by millerbryan** [cloudant/nodejs-cloudant#125]
I am unable to locate the declaration of find() to determine what the expected return value is.  I have tried:

`var foo = mydb.find({selector, stuff}, function(er, result) { more stuff; return result.docs;});`

but foo is always undefined.  The only alternative I see is to utilize some side effect inside the function which seems very untidy.  Am I missing the obvious?

**Comment by millerbryan** [comment]
My question regarding the declaration of find() remains.  However, I believe I see why foo appears to be undefined.  In my case, I am calling two find() commands in consecutive order.  Since they are asynchronous the second call is made before the first can return and assign a value to "foo".

**Comment by glynnbird** [comment]
Hi @millerbryan. Thanks for the question. The Cloudant library follows the pattern of the "nano" library - it doesn't use Promises, it uses the callback mechanism to return data to the caller.

The definition of the `find` function is here FYI: https://github.com/cloudant/nodejs-cloudant/blob/master/cloudant.js#L109

As you can see there is no return value, and as you rightly state, the calls are asynchronous so the return value from find is "returned" in the callback function. To chain to queries together in series you could do this:

```js
db.find({selector:{name:'Alice'}}, function(er, result) {
  // "result" contains the result of the first query
  db.find({selector:{name:'Rita'}}, function(er, result2) {
    // result2 contains the result of the second query
  });
}
```

Or try the async library (https://github.com/caolan/async) - `async.series` can make sequential tasks easier to manage.

**Comment by millerbryan** [comment]
Hi @glynnbird.  Thank you for the quick response.   I searched for "find(" but there must have been a disturbance in the Matrix or I noticed I needed another coffee.  Having the declaration is helpful.

I am a big fan of the async library but I think in this case I can use chaining.  I doubt I will need to ever go more than 3 levels deep.

Thank you for your work on this package.  It makes using Cloudant on Node so much cleaner.  Hopefully, it will make its way into the Cloudant "getting started" docs...
